.circleci/config.yml

```diff
@@ -0,0 +1,73 @@
+version: 2.1
+jobs:
+  misc-doxygen:
+    docker:
+      - image: bebehei/dunst:misc-doxygen-dev
+    environment:
+      SYSTEMD: 0
+      SERVICEDIR_DBUS: /tmp/none
+      SERVICEDIR_SYSTEMD: /tmp/none
+      PKG_CONFIG: echo
+    steps:
+      - checkout
+      - run: make -j doc-doxygen
+
+  compileandtest:
+    environment:
+      CFLAGS: -Werror
+    parameters:
+      distro:
+        type: string
+    docker:
+      - image: bebehei/dunst:<<parameters.distro>>-dev
+    steps:
+      - checkout
+      - run: make -j all dunstify
+      - run: make -j test-valgrind
+      - run: make -j clean
+      - run: make -j test-coverage
+
+workflows:
+  version: 2
+  build-in-docker:
+    jobs:
+      - misc-doxygen
+      - compileandtest:
+          name: Alpine
+          distro: alpine
+      - compileandtest:
+          name: Debian Stretch
+          distro: debian-stretch
+          requires:
+            - misc-doxygen
+            - Alpine
```

Comment:
Why test on Alpine first and then on everything else? I'd say it's more logical to do the initial tests on a distro that uses libc (debian or ubuntu preferably) and have alpine as one of the secondary builds after.

Reply:
Because it's the smallest image. Our build times are mainly controlled by the size of the docker image.

So you'll get the fastest response.

And if you wouldn't choose Alpine first, the build would fail too, but just in a later stage.